### PR TITLE
Forum thread editable by author

### PR DESCRIPTION
**Pull request by ondratra**
Solves subtask of #766:
- Make thread title editable by author.

To minimize merge conflict potential, this PR assumes #867 will be merged first.

### CODE REVIEW

**Comment by ondratra** [comment]
There is an inconsistency between editing the post's text and thread's title. Post saves a history of changes to the text while thread does not. How will we unite this behavior?

**Comment by bedeho** [comment]
Neither should store history of any kind, just compute and store hash commitment of most recent text.

**Comment by ondratra** [comment]
Ok, history was removed.

File: runtime-modules/forum/src/lib.rs [hunk 1]
```diff
@@ -929,25 +926,16 @@ decl_module! {
             // Signer does not match creator of post with identifier postId
             ensure!(post.author_id == forum_user_id, ERROR_ACCOUNT_DOES_NOT_MATCH_POST_AUTHOR);
 
-            // Update post text and record update history
+            // Update post text
             <PostById<T>>::mutate(post_id, |p| {
                 let text_hash = T::calculate_hash(&new_text);
```
Comment (bedeho) [inline runtime-modules/forum/src/lib.rs]: Seeing that this was already done before but in general, computation should not be done inside the `mutation` call if possible, only updating the value should be done in the closure.

**Review by bedeho — CHANGES_REQUESTED** [review]
Looks like post edit history was dropped properly, just a minor tweak.

**Review by bedeho — APPROVED** [review]

**Pull request merged**